### ISSUE TITLE
Using ShiftScale operation (Similar to BatchNorm) in MXNet (Customer question/inquiry)

### ISSUE THREAD

**Issue by sammieghabra** [apache/incubator-mxnet#18665]
## Description
Hi MXNet team,

My team wants to implement a layer in MXNet to implement a ShiftScale operator. This layer is similar to batchnorm - however, we want `moving_mean` and `moving_var` to be used instead of `data_mean` and `data_var` to compute the output of the layer. I see that batch norm has a flag `use_global_stats`, and in the [mxnet docs](http://beta.mxnet.io/r/api/mx.symbol.BatchNorm.html), it seems that setting this flag to be true would be something similar to what I'm trying to do. However, upon inspecting the [batch-norm code](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cc#L260-L271), it seems that running_mean and running_var won't be updated if that flag is set to true for training. 

1. Is there a reason why from a design perspective setting this `use_global_stats` flag to be true wont update the running mean and running var? 
2. We would like to support this shift scale layer during training. So what my proposal is to do is to add another flag to the `batchNorm` operator to be "use_shift_scale", which would simply replace mean and var with running mean and running var when updating the weights. Is this something that MXNet team would be ok  with?
3. We also plan to train with more than one instance - will the running_mean and running_var parameters be the same across instances? 

Thanks
Sammie

**Comment by rondogency** [comment]
Hi, are you using Gluon API? according to the doc it is supported https://github.com/apache/incubator-mxnet/blob/v1.7.x/python/mxnet/gluon/nn/basic_layers.py#L306, if it's not then it would be a bug in Gluon

Also which MXNet version are you using?

**Comment by sammieghabra** [comment]
@rondogency 

Thank you so much for your response.

As mentioned in the description of this issue, I am aware of the `use_global_stats` flag in batch norm. However, as I stated in the description, running mean and running var do not get updated if that flag is true for [training](https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cc#L260-L271). For our case, we need those values to be updated for training. 

I am using version 1.5 of MXNet

**Comment by rondogency** [comment]
can confirm currently MXNet doesn't support both update global running mean/var and use running mean/var to calculate grads in one pass of Batchnorm, while Keras-MXNet can support it by passing mean/var as arguments to Batchnorm as a workaround

adding the support would still have constraints, especially cudnn doesn't support it so it cannot run with the best performance; some other frameworks like caffe doesn't support this either

it can be implemented by modifying batch_norm.cc/cu and add "use_shift_scale" option to control it

**Comment by rondogency** [comment]
@zhreshold @sxjscience FYI

**Comment by ChaiBapchya** [comment]
@rondogency and this is for training right? for inference MXNet supports running mean/var right?

**Comment by ChaiBapchya** [comment]
Discussed offline. This one is for training.
